File: src/test/java/Pages/HomePage.java
package Pages;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.AriaRole;

public class HomePage {
    public Locator buttonGeti;
    public Locator buttonAcceptCookies;
    public Locator buttonStayOnThisSide;

    public HomePage(Page page){
        buttonGeti = page.locator("(//strong[contains(text(),'Meet Geti')]//parent::a//parent::div[@class='custom-button '])[1]");
        buttonAcceptCookies = page.getByLabel("Accept");
        buttonStayOnThisSide = page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Stay on this Site"));

    }
}

File: tests/pageobjects/Homepage.js
class Homepage {

    //Selectors
    constructor(page){
        this.page = page;
        this.btnAcceptCookies = page.locator("#accept");
        this.btnStayOnThisSite = page.locator("//button[@id='geoRedirectStayBtn']");
        this.btnMeetGeti = page.locator("(//a[@aria-label='<strong>Meet Geti</strong>'])[1]");
    }

}
module.exports = { Homepage };
